packages/launcher/lib/utils.ts

```diff
@@ -1,12 +1,33 @@
 import execa from 'execa'
 import cp from 'child_process'
+import os from 'os'
 import Bluebird from 'bluebird'
 
 // export an object for easy method stubbing
 export const utils = {
   execa,
+  spawnWithArch: <T extends cp.SpawnOptionsWithStdioTuple<cp.StdioNull, cp.StdioPipe, cp.StdioPipe>>(cmd: string, args: string[], opts: T) => {
+    if (!(os.platform() === 'darwin' && os.arch() === 'arm64')) {
+      return cp.spawn(cmd, args, opts)
+    }
+
+    // when Cypress spawns processes on Apple macOS with a Intel process as Cypress's parent,
+    // macOS will launch the rosetta-interpreted version (slow) by default. this wraps the
+    // spawn options to prefer arm64 over x86_64 (rosetta)
+    return cp.spawn(
```

Comment:
can you add a comment here...

> Otherwise if we're on OSX and the binary was built in arm64 then we need to signal to OSX to prefer launching browsers with arm64. There is a situation where even though the binary was built in arm64, if the user upgraded their mac and currently has a x86_64 built version of node, then the parent CLI process that launched the binary will be x86_64 and OSX will not launch the browser using arm64. We use `arch` to tell the OS to prefer the arm64 version. For users who are already using arm64 versions of node, this won't have any effect, so it's safe to do in both situations

Reply:
3d1badc4b6fe422b0e60d07f743a24d69b6aed78